kivy/uix/image.py

```diff
@@ -250,42 +290,70 @@ def texture_update(self, *largs):
         if not self.source:
             self._clear_core_image()
             return
+
+        self._loops = 0
         source = resource_find(self.source)
         if not source:
             Logger.error('Image: Not found <%s>' % self.source)
             self._clear_core_image()
             return
+        core_durations = self.durations
         if self._coreimage:
             self._coreimage.unbind(on_texture=self._on_tex_change)
+            self._coreimage.unbind(on_durations_done=self._on_durations_done)
+            core_durations = None
         try:
             self._coreimage = image = CoreImage(
                 source,
                 mipmap=self.mipmap,
                 anim_delay=self.anim_delay,
                 keep_data=self.keep_data,
-                nocache=self.nocache
+                nocache=self.nocache,
+                durations=core_durations,
+                auto_anim_delay=self.auto_anim_delay,
             )
         except Exception:
             Logger.error('Image: Error loading <%s>' % self.source)
             self._clear_core_image()
             image = self._coreimage
         if image:
             image.bind(on_texture=self._on_tex_change)
+            image.bind(on_durations_done=self._on_durations_done)
             self.texture = image.texture
 
     def on_anim_delay(self, instance, value):
         if self._coreimage is None:
             return
         self._coreimage.anim_delay = value
-        if value < 0:
+        if value < 0 and not(self.auto_anim_delay or self.durations):
             self._coreimage.anim_reset(False)
 
+    def on_auto_anim_delay(self, instance, value):
+        if self._coreimage is None:
```

Comment:
Can be replace with:
```
if self._coreimage:
    self._coreimage.auto_anim_delay = value
```

No need to test specifically for `None` and it saves one line.